source-map.bs

```diff
@@ -590,6 +611,65 @@ as per [=ECMAScript Lexical Grammar=]. [=Mapping=] entries must point to either:
     [=Comment=], [=HashbangComment=], [=StringLiteral=], [=Template=],
     [=TemplateSubstitutionTail=], [=WhiteSpace=] and [=LineTerminator=].
 
+### [=decoded mapping/Names=] for generated JavaScript code
+
+Source map generators should create a [=decoded mapping|mapping=] entry with a
+[=decoded mapping/name=] field for a JavaScript token, if
+
+1. The original source language construct maps semantically to the generated
+    JavaScript code.
+
+1. The original source language construct has a name.
+
+Then the [=decoded mapping/name=] of the [=decoded mapping|mapping=] entry should be
+the name of the original source language construct. A [=decoded mapping|mapping=] with
+a non-null [=decoded mapping/name=] is called a <dfn>named mapping</dfn>.
+
+Example: A TypeScript function compiled to a JavaScript arrow function, or a
```

Comment:
In general, TypeScript does not transpile a `function` to an arrow function in any case. Instead, we might transpile an arrow function with type information to an arrow function without, or an arrow function to a `function` as part of down-leveling to ES5.

Reply:
Replaced with "A minifier renaming functions and variables or removing function names
from immediately invoked function expressions. "